src/libraries/ui/MigrateToGaslessModal.tsx

```diff
@@ -418,6 +169,274 @@ function MigrateToGasless() {
 			/>
 		</>
 	)
+
+	// useContexts defined here
+	const { waitForScwAddress, webwallet } = useContext(WebwalletContext)!
+	const { subgraphClients } = useContext(ApiClientsContext)!
+
+	// useStates start here
+	const [isOpen, setIsOpen] = useState(false)
+	const [isConnectWalletModalOpen, setIsConnectWalletModalOpen] = useState(false)
+	const [networkModalStep, setNetworkModalStep] = useState<number>()
+	const [transactionHash, setTransactionHash] = useState<string>()
+	const [shouldMigrate, setShouldMigrate] = useState<{state: number, chainId?: SupportedChainId}>()
+	const [ownedWorkspaces, setOwnedWorkspaces] = useState<GetWorkspaceMembersQuery['workspaceMembers'][number]['workspace'][]>([])
+
+	// custom hooks
+	const { address: walletAddress } = useAccount()
+	const { data: signer } = useSigner()
+	const { chain: walletChain } = useNetwork()
+	const { switchNetwork } = useSwitchNetwork()
+	const router = useRouter()
+	const toast = useToast()
+	const workspaceContract = useQBContract('workspace', walletChain?.id as SupportedChainId, false)
+
+	const { results, fetchMore } = useMultiChainQuery({
+		useQuery: useGetProfileDetailsQuery,
+		options: {
+			variables: {
+				actorId: walletAddress ?? ''
+			}
+		}
+	})
+
+	const { results: ownedWorkspacesResults, fetchMore: fetchMoreOwnedWorkspaces } = useMultiChainQuery({
+		useQuery: useGetWorkspaceMembersQuery,
+		options: {
+			variables: {
+				actorId: walletAddress ?? ''
+			}
+		}
+	})
+
+	// variables required
+	const isMigrateYes = router?.query?.migrate === 'yes'
+
+	// functions start here
+	const onClose = (done?: boolean) => {
+		// if we're done
+		// reload the page to reflect the latest changes
+		if(done) {
+			window.location.href = window.location.pathname
+		} else {
+			const newQuery = { ...router.query }
+			delete newQuery.migrate
+
+			router.push({ query: newQuery })
+			setIsOpen(false)
+		}
+	}
+
+	const migrate = async() => {
+		try {
+			if(!walletAddress) {
+				setIsConnectWalletModalOpen(true)
+				return
+			}
+
+			if(!walletChain || !shouldMigrate || !switchNetwork || !webwallet) {
+				return
+			}
+
+			if(!(walletChain?.id in ALL_SUPPORTED_CHAIN_IDS)) {
```

Comment:
you don't need optional chaining if you have already checked for truthiness already.